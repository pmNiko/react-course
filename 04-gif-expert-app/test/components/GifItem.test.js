import { render, screen } from "@testing-library/react"
import {GifItem}  from "../../src/components/GifItem.jsx"


describe('Pruebas en el componente <GifItem />', () => {
    const imageTest = {
        title: 'Valorant',
        url: 'https://media2.giphy.com/media/CRAYdoKFzxPeqj6Ncg/giphy-downsized-medium.gif?cid=d3ecfa3ag70cq956flr6qw8ajvhv2l7jcwhj4xse5cvflu32&ep=v1_gifs_search&rid=giphy-downsized-medium.gif&ct=g'
    }

    test('Snapshot del componente', () => { 
       const { container } = render(<GifItem {...imageTest} />);

       expect(container).toMatchSnapshot();
    })    

    test('Debe mostrar la imagen con la url y el alt indicado', () => { 
        render(<GifItem {...imageTest} />);
        const { src, alt } = screen.getByRole('img');

        expect( src ).toBe(imageTest.url);
        expect( alt ).toBe(imageTest.title);
     })

    test('Debe mostrar el titulo pasado por props', () => { 
        render(<GifItem {...imageTest} />);
       
        expect( screen.getByText(imageTest.title) ).toBeTruthy();
     })
});