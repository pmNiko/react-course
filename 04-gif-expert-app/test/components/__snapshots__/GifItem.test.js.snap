// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifItem /> Snapshot del componente 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Valorant"
      src="https://media2.giphy.com/media/CRAYdoKFzxPeqj6Ncg/giphy-downsized-medium.gif?cid=d3ecfa3ag70cq956flr6qw8ajvhv2l7jcwhj4xse5cvflu32&ep=v1_gifs_search&rid=giphy-downsized-medium.gif&ct=g"
    />
    <p>
      Valorant
    </p>
  </div>
</div>
`;
